stubs/CoreGenericIterators.phpstub

```diff
@@ -376,6 +377,13 @@ class CachingIterator extends IteratorIterator implements OuterIterator , ArrayA
      * @psalm-ignore-nullable-return
      */
     public function key() {}
+
+    /**
+     * @psalm-if-this-is self<mixed, mixed, 256>
```

Comment:
This needs another `mixed` template. Should probably add a couple of tests somewhere to make sure it's working as intended. Might as well wait to see if @orklah wants to go this route though so you don't waste time only to undo it all.

Reply:
Ok I'll wait a bit before revisiting this.

Reply:
Yep, missed that one, it would need the flags template added so it can extend with the template.

Reply:
Don't forget to remove this part (sorry for the extra work, I'm sometimes too much of a perfectionist :smile:).

Reply:
No worries, I'm also perfectionist and I fully agree. Going to amend my last commit in a minute.